astropy_bot/autolabel.py

```diff
@@ -0,0 +1,24 @@
+from os import path, sep
+from baldrick.plugins.github_pull_requests import pull_request_handler
+
+
+@pull_request_handler
+def autolabel_subpackages(pr_handler, repo_handler):
+    files = pr_handler.get_modified_files()
+    all_labels = repo_handler.get_all_labels()
+    existing_labels = set(pr_handler.labels)
+
+    for file in files:
+        subdir = path.dirname(file)
+
+        if subdir:
+            root, *subpkg = subdir.split(sep)
+
+            if subpkg:
+                dot_name = '.'.join(subpkg)
+                if dot_name in all_labels and dot_name not in existing_labels:
+                    existing_labels.add(dot_name)
+
+    pr_handler.set_labels(existing_labels)
```

Comment:
I don't know how all these different handlers work in the background. Will they be in race condition with each other, @astrofrog or @Cadair ?

Reply:
This would be addressed by only running when pull requests are "opened"